### ISSUE TITLE
internal: Missing key in the documents database..

### ISSUE THREAD

**Issue by thijndehaas** [meilisearch/meilisearch#3664]
**Describe the bug**
The `internal: Missing key in the documents database..` seems to be back again. 

I can index the products without problems but as soon as I submit them again all tasks fail. 

We have 5 websites running on `1.1.0` they all have this problem. 
We have 8 websites running on `0.30.5` and they don't have this problem. The data we use before/after 0.30.5 is the same.

**Screenshots**
Meilisearch 0.30.5
![afbeelding](https://user-images.githubusercontent.com/18503351/231282420-72b2123f-105c-4b87-a826-7dcfa16f4c59.png)
Meilisearch 1.1.0
![afbeelding](https://user-images.githubusercontent.com/18503351/231282358-80719c61-e6d6-429a-9340-05f965ec040e.png)

**Meilisearch version:**
`v1.1.0`

**Additional context**
Debian 5.10.70-1
128GB ram
1.5TB disk

**Last update and delete tasks**
```json
[
    {
        "uid": 757,
        "indexUid": "indexname",
        "status": "failed",
        "type": "documentAdditionOrUpdate",
        "canceledBy": null,
        "details": {
            "receivedDocuments": 8573,
            "indexedDocuments": 0
        },
        "error": {
            "message": "internal: Missing key in the documents database..",
            "code": "internal",
            "type": "internal",
            "link": "https:\/\/docs.meilisearch.com\/errors#internal"
        },
        "duration": "PT0.016145437S",
        "enqueuedAt": "2023-04-11T20:38:47.258877067Z",
        "startedAt": "2023-04-11T20:38:47.25921795Z",
        "finishedAt": "2023-04-11T20:38:47.275363387Z"
    },
    {
        "uid": 756,
        "indexUid": "indexname",
        "status": "failed",
        "type": "documentDeletion",
        "canceledBy": null,
        "details": {
            "providedIds": 89386,
            "deletedDocuments": 0
        },
        "error": {
            "message": "internal: Missing key in the documents database..",
            "code": "internal",
            "type": "internal",
            "link": "https:\/\/docs.meilisearch.com\/errors#internal"
        },
        "duration": "PT0.055855178S",
        "enqueuedAt": "2023-04-11T20:38:46.71434441Z",
        "startedAt": "2023-04-11T20:38:46.75512343Z",
        "finishedAt": "2023-04-11T20:38:46.810978608Z"
    }
]
```

**Comment by phaistonian** [comment]
Same here.

<img width="889" alt="image" src="https://user-images.githubusercontent.com/796530/231383246-96bf85b1-8b00-4743-a839-1dde921f5ae2.png">

**Comment by dureuill** [comment]
Hello @thijndehaas and @phaistonian,

This is most definitely a bug in the engine, most certainly related to soft-deletion of documents again.

1. Can you confirm that it happens when sending documents with changes to an index?
2. Can you share the sequence of actions (e.g. requests you made to the server) that lead to the error?
3. Can you share the settings of an index and a dataset that reproduces the issue?

If you need to share the dataset privately, you can reach me on [discord](https://discord.gg/meilisearch) and send a direct message to me or by email at my Meilisearch address in my GitHub profile.

Thank you for the report, we will investigate when we can reproduce

**Comment by phaistonian** [comment]
@dureuill: Thank you for your quick response.

1. Yes
2. Delete tracked entries from index (usually a couple hundred and then add/update the new ones, usually 1.5m or so.
3. Well, the current dataset for this index is 85GB so sending this will be an issue - or more than one :)

My "theory" is that since #2 is invoked by a cron running every 8 hours, some of the items meant to be either deleted or added are not yet processed, so the new scheduled task somehow conflicts with the enqueued tasks.

A rather simplistic theory, but that might be it.

**Comment by dureuill** [comment]
Thank you for the response. We have a working hypothesis of what indexing change might have caused this regression in v1.1. I will try and come up with a prototype that reverts that change for you to test and see if it fixes the issue. I'll keep you posted.

**Comment by dureuill** [comment]
@phaistonian , @thijndehaas, we just released a prototype to test our hypothesis that the autobatching of document additions and deletions is responsible for the issue. If we are correct it should fix the issue.

How to run the prototype? You need to start from a fresh new database (remove the previously used `data.ms`) and use the following Docker image:

```
docker run -it --rm -p 7700:7700 -v $(pwd)/meili_data:/meili_data getmeili/meilisearch:prototype-disable-autobatch-addition-deletion-0
```

You can then try to reproduce the failing operations and check they no longer fail.

⚠️ We do NOT recommend using this prototype in production. This is only for test purposes.

If the prototype succeeds in fixing the issue, we might release it as a patch release or work on a fix for the batching of document addition and deletion.

Thanks in advance for your involvement. We hope this will resolve the issue.

**Comment by phaistonian** [comment]
@dureuill  We are on it - I will get back to you with any findings.

**Comment by phaistonian** [comment]
@dureuill The test was successful - I am told. It seems that the new version does resolve the issue.

ETA on pushing it live?:)

**Comment by dureuill** [comment]
As a quick update, we'll release soon a [v1.1.1](https://github.com/meilisearch/meilisearch/milestone/49) containing the hotfix that disables autobatching of additions and deletions as well as an unrelated bugfix.

**Comment by thijndehaas** [comment]
Is there an ETA when v1.1.1 will be released? Or can I manually create a file like `meilisearch-linux-amd64`? Currently all my indexes seem to fail within a day after re-creating them. I'm now recreating the index and swapping it every day to avoid to much incorrect/outdated data 😅.

**Comment by dureuill** [comment]
Hello @thijndehaas, @phaistonian 👋

We just released [v1.1.1](https://github.com/meilisearch/meilisearch/releases/tag/v1.1.1) that contains the fix we previously published as a prototype.

I'm closing this issue as it should now be fixed 😀

**Comment by phaistonian** [comment]
Thank you @dureuill  :-)
